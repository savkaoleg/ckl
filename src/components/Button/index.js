// @flow
import React from 'react'
import './index.scss'

const Button = ({
  text,
  onClick,
  raised,
  primary,
  warn
}: {
  text: string,
  onClick: Function,
  raised?: boolean,
  primary?: boolean,
  warn?: boolean
}) => (
  <button
    type="button"
    className={`md-btn md-btn-ripple ${raised ? 'md-btn-raised' : ''} ${
      primary ? 'md-btn-primary' : ''
    } ${warn ? 'md-btn-warn' : ''}`}
    onClick={onClick}
  >
    {text}
  </button>
)

export default Button
