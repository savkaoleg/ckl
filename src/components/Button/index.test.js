import React from 'react'
import expect from 'expect'
import { shallow } from 'enzyme'
import Button from './index'

describe('Button', () => {
  it('Renders correctly', () => {
    const component = shallow(<Button />)

    expect(component).toMatchSnapshot()
  })

  it('raised prop', () => {
    const component = shallow(<Button raised />)
    expect(component.hasClass('md-btn-raised')).toEqual(true)
  })

  it('primary prop', () => {
    const component = shallow(<Button primary />)
    expect(component.hasClass('md-btn-primary')).toEqual(true)
  })

  it('primary warn', () => {
    const component = shallow(<Button warn />)
    expect(component.hasClass('md-btn-warn')).toEqual(true)
  })
})
