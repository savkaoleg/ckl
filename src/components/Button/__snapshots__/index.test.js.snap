// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders correctly 1`] = `
<button
  className="md-btn md-btn-ripple   "
  type="button"
/>
`;
